src/chttpd/test/eunit/chttpd_util_test.erl

```diff
@@ -0,0 +1,94 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(chttpd_util_test).
+
+-include_lib("couch/include/couch_eunit.hrl").
+-include("chttpd_test.hrl").
+
+setup() -> ok.
+
+teardown(_) -> ok.
+
+chttpd_util_config_test_() ->
+	{
+		"chttpd util config tests",
+		{
+			setup,
+			fun test_util:start_couch/0,
+			fun test_util:stop_couch/1,
+			{
+				foreach,
+				fun setup/0,
+				fun teardown/1,
+				[
+					?TDEF_FE(get_chttpd_config_with_undefined_option),
+					?TDEF_FE(get_chttpd_config_with_httpd_option),
+					?TDEF_FE(get_chttpd_config_with_chttpd_option),
+					?TDEF_FE(get_chttpd_config_with_chttpd_option_which_moved_from_httpd),
+					?TDEF_FE(get_chttpd_auth_config_with_undefined_option),
+					?TDEF_FE(get_chttpd_auth_config_with_couch_httpd_auth_option),
+					?TDEF_FE(get_chttpd_auth_config_with_chttpd_auth_option_which_moved_from_couch_httpd_auth)
+				]
+			}
+		}
+	}.
+
+get_chttpd_config_with_undefined_option(_) ->
+	?assertEqual(undefined, chttpd_util:get_chttpd_config(undefined_option)),
+	?assertEqual(default, chttpd_util:get_chttpd_config(undefined_option, default)),
+	?assertEqual(123, chttpd_util:get_chttpd_config(undefined_option, 123)),
+	?assertEqual(0.2, chttpd_util:get_chttpd_config(undefined_option, 0.2)),
+	?assert(chttpd_util:get_chttpd_config(undefined_option, true)),
+	?assertNot(chttpd_util:get_chttpd_config(undefined_option, false)),
+	?assertEqual("abc", chttpd_util:get_chttpd_config(undefined_option, "abc")),
+	?assertEqual("", chttpd_util:get_chttpd_config(undefined_option, "")).
+
+get_chttpd_config_with_httpd_option(_) ->
+	?assertEqual("0", chttpd_util:get_chttpd_config(port)),
+	?assertEqual("127.0.0.1", chttpd_util:get_chttpd_config(bind_address)),
+	?assertEqual("0", chttpd_util:get_chttpd_config(port, "8080")),
+	?assertEqual("127.0.0.1", chttpd_util:get_chttpd_config(bind_address, "255.255.255.255")).
```

Comment:
We don't want to get the httpd socket options via the chttpd_util shim functions. Options which pertain to the socket connection would stay in httpd (port, bind_address, socket and server settings).